metricflow/plan_conversion/instance_converters.py

```diff
@@ -64,41 +68,94 @@ def __init__(
         self,
         table_alias: str,
         column_resolver: ColumnAssociationResolver,
+        spec_output_order: Sequence[InstanceSpec] = (),
         output_to_input_column_mapping: Optional[OrderedDict[str, str]] = None,
     ) -> None:
         """Initializer.
 
         Args:
             table_alias: the table alias to select columns from
             column_resolver: resolver to name columns.
+            spec_output_order: If specified, order the output columns for instances according to the given order of the
```

Comment:
It feels odd to put this on an instance converter when that requires a lot of denormalization. We have to pass in the order of specs in addition to the instance set (which already has all the specs attached) because the instance set doesn't allow you to specify the order across different types of group bys. Then we have to return the columns twice, too. Plus it feels weird to pass `spec_output_order` into the initializer when you won't even have the instances until they get passed into the `transform()` method.

What if you make `columns_in_order` a method on `SelectColumnSet` instead and do the sorting there? Could move the `spec_output_order` parameter to that method and default to the standard sorting we want.

Reply:
> It feels odd to put this on an instance converter when that requires a lot of denormalization

I think I get what you mean by `denormalization`, but can you clarify what you mean?

> What if you make columns_in_order a method on SelectColumnSet instead and do the sorting there? Could move the spec_output_order parameter to that method and default to the standard sorting we want.

Yeah, that makes sense, but it does add more exposed classes since this performs the mapping. I've added a few commits that do this.

Reply:
By `denormalization` I mean we're including all the specs twice on both the inputs and the outputs. For example, if this converter was called for a `metric_time__day` instance:
- The `metric_time` spec already included on the input instance set, and now it's duplicated in the `spec_output_order` param
- The `metric_time__day` column is already returned in `SelectColumnSet.time_dimension_columns`, and now it's duplicated in `SelectColumnSet.columns_in_order`.

The changes you made look good to me!